libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -139,9 +183,52 @@ void DocStringAnalyser::handleCallable(
 	StructurallyDocumentedAnnotation& _annotation
 )
 {
-	static set<string> const validTags = set<string>{"author", "dev", "notice", "return", "param"};
+	static set<string> const validTags = set<string>{"author", "dev", "notice", "return", "param", "inheritdoc"};
 	parseDocStrings(_node, _annotation, validTags, "functions");
 	checkParameters(_callable, _node, _annotation);
+
+	if (CallableDeclaration const* baseFunction = resolveInheritDoc(_callable.annotation().baseFunctions, _node, _annotation))
+		copyMissingTags(_annotation, baseFunction->annotation());
+	else if (
+		_annotation.docTags.empty() &&
+		_callable.annotation().baseFunctions.size() == 1 &&
+		parameterNamesEqual(_callable, **_callable.annotation().baseFunctions.begin())
+	)
+		copyMissingTags(_annotation, (*_callable.annotation().baseFunctions.begin())->annotation());
+}
+
+CallableDeclaration const* DocStringAnalyser::resolveInheritDoc(
+	std::set<CallableDeclaration const*>& _baseFuncs,
+	StructurallyDocumented const& _node,
+	StructurallyDocumentedAnnotation& _annotation
+)
+{
+	switch (_annotation.docTags.count("inheritdoc"))
+	{
+		default:
+			m_errorReporter.docstringParsingError(
+				5142_error,
+				_node.documentation()->location(),
+				"Documentation tag @inheritdoc can only reference one contract."
+			);
+			return nullptr;
+		case 0:
+			return nullptr;
+		case 1:
+			string const targetContractName = _annotation.docTags.find("inheritdoc")->second.content;
```

Comment:
This can contain spaces, can it?

Reply:
I did a short test and it seems it's trimmed. A space before/after is ignored.